### DISCUSSION TITLE
Restore fails, if lock is disabled, but lock file exists

### DISCUSSION THREAD

**Discussion by uladz-zubrycki** [NuGet/Home#10599] [Q&A] [answered]
#### Description ####
Hi! It's more like a question than an issue actually. 

If I have lock mode disabled, but `packages.lock.json` file exists for a project, restore will fail, why is that?

```
NU1005 Invalid restore input where RestorePackagesWithLockFile property is set to false but a packages lock file exists at {project-path}\packages.lock.json.	
```			

#### Environment ####
Microsoft Visual Studio Community 2019
Version 16.8.3
VisualStudio.16.Release/16.8.3+30804.86
Microsoft .NET Framework
Version 4.8.03752

**Comment by donnie-msft** [comment]
Hi! Given this is a question about using the product, I'm moving it to the Discussions area.

**Comment by nkolev92** [comment, marked as answer]
> If I have lock mode disabled

The error message `Invalid restore input where RestorePackagesWithLockFile property is set to false but a packages lock file exists` suggests that you are trying to disable lock files altogether, and not locked mode. 

The opt in experience for lock files is the existence of a lock file or that property being true. 

Setting that property to true + having a lock file is not usually a state you'd want to be in on purpose. 
If you want to disable lock files, deleting the file is a necessity. 

That being said, I think you are looking for `RestoreLockedMode` which is what controls the locking functionality.

**Reply by uladz-zubrycki** [reply]
> The opt in experience for lock files is the existence of a lock file

This sheds light on the decision a bit, thanks. Was mistakenely sure that msbuild prop is a must for lock files support. 

I've actually been trying to disable lock files regeneration temporarily without deleting the files themselves, so this error appeared as impediment. Not actual anymore, as idea got to be impossible, but got curious why decision to report an error, when there is no error was made. 

Thanks.

**Reply by kartheekp-ms** [reply]
AFAIK, this behavior is by design, As per [doc](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#enabling-lock-file), `Once a project has packages.lock.json file in its root directory, the lock file is always used with restore even if the property RestorePackagesWithLockFile is not set. So another way to opt-in to this feature is to create a dummy blank packages.lock.json file in the project's root directory.`
